### DISCUSSION TITLE
React with Webpack & SignalR

### DISCUSSION THREAD

**Discussion by HeckezMX** [dotnet/aspnetcore#27334] [Q&A]
Hi everyone!.
I need your help. I have a react project with the signalR integration. My first implementation was with `react-scripts` builder and works perfect, but now I'm migrating to **Webpack**, but when Webpack tries to build the app throws me some errors related with the Http connections like...

`new HttpConnection`
`HttpConnection../node_modules/@microsoft/signalr/dist/esm/HttpConnection.js.HttpConnection.resolveUrl`
`HubConnectionBuilder../node_modules/@microsoft/signalr/dist/esm/HubConnectionBuilder.js.HubConnectionBuilder.build`
`new SignalRService`

Do you know some configuration in Webpack to avoid this errors?

**Comment by BrennanConroy** [comment]
Are you missing some text for those errors? They don't look like errors in their current form.

We have a tutorial that sets up a project using WebPack which might help https://docs.microsoft.com/aspnet/core/tutorials/signalr-typescript-webpack?view=aspnetcore-3.1&tabs=visual-studio

**Reply by HeckezMX** [reply]
Hi @BrennanConroy, thanks for you feedback. Yeah I'm only put some messages, but, this are the errors in console.

![webpackErrors](https://user-images.githubusercontent.com/20602485/97642794-b21adf00-1a0b-11eb-82f2-07b2bec8933b.png)

I followed the error lines and some are related with signalR. This is part of my webpack configuration.

![image](https://user-images.githubusercontent.com/20602485/97643062-56048a80-1a0c-11eb-9059-974eb29c39a5.png)

Note: We're not using TS in the React project.

**Reply by BrennanConroy** [reply]
~~Are you using the @aspnet/signalr npm package?~~
Also, what are you passing into `withUrl`?

**Comment by HeckezMX** [comment]
I'm using this package @microsoft/signalr

![image](https://user-images.githubusercontent.com/20602485/97646629-c82d9d00-1a15-11eb-9d02-069dbe56e371.png)

And I'm passing an endpoint that was created as Azure function. For example for mi local env. `http://127.0.0.1:7071/api`

**Reply by BrennanConroy** [reply]
Hmm, I'm not sure then. Could you create a repro app, or if it's easy the steps to make an app?

**Reply by HeckezMX** [reply]
@BrennanConroy Sorry  I was on vacations. Yesterday I retake this and I solved this issue removing the `/node_modules/` folder and running again the command `npm install`. Thanks for your support tracking this.
